src/platform/Darwin/KeyValueStoreManagerImpl.mm

```diff
@@ -202,7 +202,7 @@ - (instancetype)initWithContext:(nonnull NSManagedObjectContext *)context
             }
 
             if (value != nullptr) {
-                memcpy(value, item.value.bytes, std::min(item.value.length, value_size));
+                memcpy(value, item.value.bytes, std::min(static_cast<size_t>(item.value.length), value_size));
```

Comment:
nit: nstead of casting value.length you could `std::min<size_t>(item.value.length, value_size)` to force which min version is used. This way you do not depend on value_size being exactly size_t.

Reply:
Yeah that's even better. Updated it!